packages/next/lib/eslint/writeOutputFile.ts

```diff
@@ -0,0 +1,45 @@
+import { promises as fs } from 'fs'
+import path from 'path'
+import * as Log from '../../build/output/log'
+import isError from '../../lib/is-error'
+
+/**
+ * Check if a given file path is a directory or not.
+ * @param {string} filePath The path to a file to check.
+ * @returns {Promise<boolean>} `true` if the path is a directory.
+ */
+async function isDirectory(filePath: string): Promise<boolean> {
+  try {
+    return (await fs.stat(filePath)).isDirectory()
+  } catch (error) {
+    if (
+      isError(error) &&
+      (error.code === 'ENOENT' || error.code === 'ENOTDIR')
+    ) {
+      return false
+    }
+    throw error
+  }
+}
+/**
+ * Create a file with eslint output data
+ * @param {string} outputFile The name file that needs to be created
+ * @param {string} outputData The data that needs to be inserted into the file
+ */
+export async function writeOutputFile(outputFile: string, outputData: string) {
+  const filePath = path.resolve(process.cwd(), outputFile)
+
+  if (await isDirectory(filePath)) {
```

Comment:
Does `isDirectory` need to be a separate function instead of just checking for `fs.stat(filePath)).isDirectory()` here directly?

Reply:
So I create a separate function because `fs.stat(filePath)).isDirectory()` returns an exception when the path is not a directory if I put `fs.stat(filePath)).isDirectory()` directly I always receive this message in my output when I tried to run the next project using a local dependency:
```shell
ENOENT: no such file or directory, stat '/Users/evertonnunes/workspace/example/report.json'
```

For me a separate function is easier to understand what is going on than trying to create another try/catch or something else, what do you think about it?



Example with a try/catch wrapping:
```ts
try {
    if ((await fs.stat(filePath)).isDirectory()) {
      Log.error(
        `Cannot write to output file path, it is a directory: ${filePath}`
      )
    }
  } catch { // the filePath is not a directory
    try {
      await fs.mkdir(path.dirname(filePath), { recursive: true })
      await fs.writeFile(filePath, outputData)
      Log.info(`The output file has been created: ${filePath}`)
    } catch {
      Log.error(`There was a problem writing the output file: ${filePath}`)
    }
  }
```

Reply:
That makes sense! Thanks for clarifying :)

Now that I understand - I definitely agree that having a separate function here makes more sense.